packages/contracts-bridge/contracts/BridgeMessage.sol

```diff
@@ -326,6 +364,50 @@ library BridgeMessage {
         return _message.slice(0, TOKEN_ID_LEN, uint40(Types.TokenId));
     }
 
+    /**
+     * @notice Retrieves the hook contract EVM address from a TransferWithHook
+     * @param _transferAction The message
+     * @return The hook contract address as bytes32
+     */
+    function evmHook(bytes29 _transferAction)
+        internal
+        pure
+        typeAssert(_transferAction, Types.TransferToHook)
+        returns (address)
+    {
+        return _transferAction.indexAddress(13);
+    }
+
+    /**
+     * @notice Retrieves the hook contract from a TransferWithHook
+     * @param _transferAction The message
+     * @return The hook contract address as an address
+     */
+    function hook(bytes29 _transferAction)
+        internal
+        pure
+        typeAssert(_transferAction, Types.TransferToHook)
+        returns (bytes32)
+    {
+        // before = 1 byte identifier
+        return _transferAction.index(1, 32);
+    }
+
+    function extraData(bytes29 _transferAction)
+        internal
+        pure
+        typeAssert(_transferAction, Types.TransferToHook)
+        returns (bytes29)
+    {
+        // anything past the end is the extradata
+        return
+            _transferAction.slice(
+                TRANSFER_LEN,
+                _transferAction.length - TRANSFER_LEN,
+                uint40(Types.ExtraData)
```

Comment:
why have a type for this?

Reply:
cleanliness, basically.

Reply:
i don't think i understand what's more clean about this

Reply:
using 0 for the type everywhere is basically unintended behavior for the lib. it's cleaner to type your typed mem views